### DISCUSSION TITLE
How can I convert slint::Image to Vec<u8>?

### DISCUSSION THREAD

**Discussion by Horbin-Magician** [slint-ui/slint#4169] [Q&A]
I use the `screenshots` crate to take a screen shot, and convert the picture into a `slint::Image`.
After that, how can I save the picture into a .png file?
Maybe I need to convert slint::Image to Vec<u8>.

**Comment by tronical** [comment]
There's currently no way to get the data out of a `slint::Image`, but as you create the `slint::Image` yourself from the data from the screenshots crate, you get an [`RgbaImage`](https://docs.rs/image/0.24.7/image/type.RgbaImage.html), and that one has a [`save_with_format`](https://docs.rs/image/0.24.7/image/struct.ImageBuffer.html#method.save_with_format) function, which you can call to save the data to a `.png` file.

**Reply by Horbin-Magician** [reply]
This means I need to keep two copies of the image in memory.
Is there really no better way?

**Reply by tronical** [reply]
When you create the `slint::Image`, you'll likely use something like [`from_rgba8`](https://docs.rs/slint/latest/slint/struct.Image.html#method.from_rgba8). That function takes a `SharedPixelBuffer`, which you must have created. Since that one is shared, you can keep a clone around of it for free. And later when you want to save it to a `.png` file, you can access the data [`as_bytes()`](https://docs.rs/slint/latest/slint/struct.SharedPixelBuffer.html#method.as_bytes) and call [`save_buffer`](https://docs.rs/image/0.24.7/image/fn.save_buffer.html) to encode and write to a file.

That way there's only a single copy of the image in memory.

**Reply by Horbin-Magician** [reply]
I get it. Thank you for your reply.
